### PR TITLE
Add home page for Linux Consumption app

### PR DESCRIPTION
**Pull request by Hazhzeng**
1. Add homepage for Linux Consumption (Only exposing api/settings, and api/deployments)
image
2. Change Kudu.Core.Environment.IsOnLinuxConsumption to static, since it derives from environment variables.
3. Fix kudulite commit path to `https://github.com/Azure-App-Service/KuduLite/commit/<sha>`, (previously incorrectly pointed to `https://github.com/ProjectKudu/Kudu/commit/<sha>`).

cc: @balag0

### CODE REVIEW

**Comment by Hazhzeng** [comment]
@sanchitmehta Please review, thanks \o/

File: Kudu.Core/Environment.cs [hunk 1]
```diff
@@ -410,7 +410,7 @@ public string RequestId
             private set;
         }
 
-        public bool IsOnLinuxConsumption
+        public static bool IsOnLinuxConsumption
```
Comment (balag0) [inline Kudu.Core/Environment.cs:413]: Is it possible to keep this non-static. (makes writing tests easier)

File: Kudu.Services.Web/Pages/Index.cshtml [hunk 2]
```diff
@@ -100,24 +101,30 @@
...
-    </ul>
-    <h3>Browse Directory</h3>
-    <ul>
-        <li>
-            <a href="deploymentlogs">Deployment Logs</a>
-        </li>
-        <li>
-            <a href="wwwroot">Site wwwroot</a>
-        </li>
+        @if (!Kudu.Core.Environment.IsOnLinuxConsumption)
```
Comment (balag0) [inline Kudu.Services.Web/Pages/Index.cshtml:104]: does cshtml support creating partial views (one for consumption and another for dedicated) to keep the logic isolated and reduce the ifs in this file.

File: Kudu.Services.Web/Pages/_Layout.cshtml [hunk 3]
```diff
@@ -65,12 +65,15 @@
                 <a class="navbar-brand" href="~/" style="padding-top: 12px;color:white !important;"><img src="/Content/Images/AppService.png" style="width: 30px; height: 30px"> Azure App Service </a>
             </div>
             <div class="collapse navbar-collapse">
-                <ul class="nav navbar-nav">
-                    <li><a class="navbar-brand" style="padding-top: 17px" href="~/Env">Environment</a></li>
-                    <li><a class="navbar-brand" style="padding-top: 17px" href="~/webssh/host">SSH</a></li>
-                    <li><a class="navbar-brand" style="padding-top: 17px" href="~/DebugConsole">Bash</a></li>
-                    <li><a class="navbar-brand" style="padding-top: 17px" href="~/api/logstream" title="If no log events are being generated the page may not load.">Log stream</a></li>
-                </ul>
+                @if (!Kudu.Core.Environment.IsOnLinuxConsumption)
```
Comment (balag0) [inline Kudu.Services.Web/Pages/_Layout.cshtml:68]: similar comment as index.cshtml. possibility of partial views?

File: Kudu.Services/LinuxConsumptionInstanceAdmin/LinuxConsumptionRouteMiddleware.cs [hunk 4]
```diff
@@ -159,6 +147,9 @@ public async Task Invoke(HttpContext context, IAuthorizationService authorizatio
 
         private bool IsRouteWhitelisted(PathString routePath)
```
Comment (balag0) [inline Kudu.Services/LinuxConsumptionInstanceAdmin/LinuxConsumptionRouteMiddleware.cs:148]: nit: rename to IsRouteAllowed

File: Kudu.Services/LinuxConsumptionInstanceAdmin/LinuxConsumptionRouteMiddleware.cs [hunk 5]
```diff
@@ -159,6 +147,9 @@ public async Task Invoke(HttpContext context, IAuthorizationService authorizatio
...
         {
+            if (IsHomePageRoute(routePath)) {
+                return true;
+            }
             return _whitelistedPathString.Any((ps) => routePath.StartsWithSegments(ps));
```
Comment (balag0) [inline Kudu.Services/LinuxConsumptionInstanceAdmin/LinuxConsumptionRouteMiddleware.cs:153]: nit: rename to _allowedPaths

**Review by balag0 — CHANGES_REQUESTED** [review]

**Comment by sanchitmehta** [comment]
LGTM!

**Review by sanchitmehta — APPROVED** [review]

**Pull request merged**